nixos/modules/services/web-apps/pr-tracker.nix

```diff
@@ -0,0 +1,315 @@
+      description = ''
+        The URL where users can download the program's source code.
+      '';
+    };
+
+    mountPath = mkOption {
+      type = with types; nullOr str;
+      example = "pr-tracker.html";
+
+      default = null;
+      description = ''
+        A "mount" path can be specified, which will be prefixed to all
+        of the server's routes, so that it can be served at a non-root
+        HTTP path.
+      '';
+    };
+
+    user = mkOption {
+      default = "pr-tracker";
+      type = types.str;
+      description = ''
+        User account under which pr-tracker runs.
+
+        ::: {.note}
+        If left as the default value this user will automatically be created
+        on system activation, otherwise you are responsible for ensuring the
+        user exists before the pr-tracker service starts.
+        :::
+      '';
+    };
+
+    group = mkOption {
+      default = "pr-tracker";
+      type = types.str;
+      description = ''
+        Group account under which pr-tracker runs.
+
+        ::: {.note}
+        If left as the default value this user will automatically be created
+        on system activation, otherwise you are responsible for ensuring the
+        user exists before the pr-tracker service starts.
+        :::
+      '';
+    };
+
+    dataDir = mkOption {
+      type = types.path;
+      example = "/home/nixos/git";
+
+      default = "/var/lib/pr-tracker";
+      description = ''
+        The data directory for pr-tracker.
+
+        ::: {.note}
+        If left as the default value this directory will automatically be created
+        before the pr-tracker server starts, otherwise you are responsible for ensuring
+        the directory exists with appropriate ownership and permissions.
+        :::
+      '';
+    };
+
+    listen = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      example = "127.0.0.1";
+      description = ''
+        The IP address to bind the socket to.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 3000;
+      description = ''
+        TCP port where pr-tracker socket listens.
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Open ports needed for the functionality of the program.
+      '';
+    };
+  };
+
+  config =
+    let
+      useClone = cfg.nixpkgsClone.managedByModule;
+
+      prestart = "${
+        pkgs.writeShellApplication {
+          name = "pr-tracker-pre";
+          runtimeInputs = [ pkgs.git ];
+
+          text = ''
+            if [ ! -d ${cfg.nixpkgsClone.cloneDir} ]; then
+                git clone https://github.com/NixOS/nixpkgs.git ${cfg.nixpkgsClone.cloneDir}
+            fi
+          '';
+        }
+      }/bin/pr-tracker-pre";
+
+      commonUnitSettings = {
+        User = cfg.user;
+        Group = cfg.group;
+
+        StateDirectory = builtins.baseNameOf cfg.dataDir;
```

Comment:
This only works if dataDir is inside /var/lib

Reply:
Hmm, what should I do about that? Make StateDirectory optional if cfg.dataDir is not default?

Reply:
The easiest way would be to just hardcode /var/lib and don't worry about. We could also do your suggestion but who really tests this and makes sure it continues to work? IMO not worth the effort.